src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double GROWING_COEFFICIENT = 1.5;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (indexIsInvalid(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index passed");
+        }
```

Comment:
Let's collapse it in one method.
```suggestion
        verifyIndexIsValid();
```

The same for all next occurrences.